src/lib/utils.ts

```diff
@@ -149,4 +150,9 @@ export const getCertData = (cert: string): ICertData => {
     certData.organizationUnit = ou[1];
   }
   return certData;
-};
\ No newline at end of file
+};
+
+export const extractPeerSenderFromRequest = (req: Request): string => {
+  const cert = ((req.socket) as TLSSocket).getPeerCertificate();
+  return getPeerID(cert.subject.O, cert.subject.OU);
```

Comment:
You may need to update custom.d.ts to avoid potential compiler issues.

Reply:
I hadn't hit compiler issues when I built and tested this code in my deployments, but I updated `custom.d.ts` for good measure and removed the unused interfaces.